### PR TITLE
Add healthcheck command

### LINKED ISSUE #1560: [Feature Request] Make /ping accessible overriding the HTTP basic authentication
Currently on Traefik v1.2.3, the `/ping` can only be accessed at `[web]` API backend.

If you configure it with TLS and HTTP basic authentication, you will also need to submit your username and password when you just want to access `/ping` for basic healthcheck. 

This is not a good idea when you are building your own Docker image with `HEALTHCHECK` instruction (e.g. `HEALTHCHECK CMD curl -f -k -u username:password https://localhost:8080/ping || exit 1`).

I hope that the `/ping` can be accessible without requiring for HTTP basic authentication at least when queried from the localhost.

### PR DESCRIPTION
**Pull request by emilevauge**
### Description

This PR adds a `traefik healthcheck` command that can be used by Docker with the new [`HEALTHCHECK`](https://docs.docker.com/engine/reference/builder/#healthcheck) instruction (see https://github.com/containous/traefik-library-image/pull/12).
The `web` provider needs to be enabled.
/cc @EtienneDufresne @vdemeester 

Fixes #1560

### CODE REVIEW

File: cmd/traefik/traefik.go [hunk 1]
```diff
@@ -97,6 +99,43 @@ Complete documentation is available at https://traefik.io`,
...
+					TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+				}
+				client.Transport = tr
+			}
+			resp, err := client.Head(protocol + "://" + traefikConfiguration.Web.Address + "/ping")
+			if err != nil {
+				fmt.Printf("Error calling healthcheck: %s\n", err)
+				os.Exit(1)
+			}
+			if resp.Status != "200 OK" {
```
Comment (timoreimann) [inline cmd/traefik/traefik.go]: I'd check `resp.StatusCode` for `http.StatusOK`.

Comment (emilevauge) [inline cmd/traefik/traefik.go]: Absolutely

File: cmd/traefik/traefik.go [hunk 2]
```diff
@@ -26,6 +27,7 @@ import (
 	"github.com/coreos/go-systemd/daemon"
 	"github.com/docker/libkv/store"
 	"github.com/satori/go.uuid"
+	"net/http"
```
Comment (vdemeester) [inline cmd/traefik/traefik.go]: The import are a little messed-up 😓 😛

Comment (emilevauge) [inline cmd/traefik/traefik.go]: Fixed

File: docs/basics.md [hunk 3]
```diff
@@ -514,6 +514,7 @@ List of Træfik available commands with description :             
 - `version` : Print version 
 - `storeconfig` : Store the static traefik configuration into a Key-value stores. Please refer to the [Store Træfik configuration](/user-guide/kv-config/#store-trfk-configuration) section to get documentation on it.
 - `bug`: The easiest way to submit a pre-filled issue.
+- `healthcheck`: Calls traefik `/ping` to check health
```
Comment (ldez) [inline docs/basics.md]: Add a dot at the end of the sentence.

File: cmd/traefik/traefik.go [hunk 4]
```diff
@@ -97,6 +99,43 @@ Complete documentation is available at https://traefik.io`,
...
+			resp, err := client.Head(protocol + "://" + traefikConfiguration.Web.Address + "/ping")
+			if err != nil {
+				fmt.Printf("Error calling healthcheck: %s\n", err)
+				os.Exit(1)
+			}
+			if resp.StatusCode != http.StatusOK {
+				fmt.Printf("Bad healthcheck status: %s\n", resp.Status)
+				os.Exit(1)
+			}
+			fmt.Printf("OK: %s\n", resp.Request.URL.String())
```
Comment (emilevauge) [inline cmd/traefik/traefik.go]: Not sure to understand, `fmt.Printf("OK: %s\n", resp.Request.URL.String())` -> `fmt.Printf("OK: %s\n", resp.Request.URL)` ? Why ?

File: cmd/traefik/traefik.go [hunk 5]
```diff
@@ -97,6 +99,43 @@ Complete documentation is available at https://traefik.io`,
...
 	}
 
+	healthcheckCmd := &flaeg.Command{
+		Name:                  "healthcheck",
+		Description:           `Calls traefik /ping to check health (web provider must be enabled)`,
+		Config:                traefikConfiguration,
+		DefaultPointersConfig: traefikPointersConfiguration,
+		Run: func() error {
+			if traefikConfiguration.Web == nil {
+				fmt.Printf("Please enable the web provider to use healtcheck.\n")
```
Comment (emilevauge) [inline cmd/traefik/traefik.go]: I was thinking of homogeneity, but meh, I can change this if you want.

File: cmd/traefik/traefik.go (hunk 4 above)
Comment (ldez) [inline cmd/traefik/traefik.go]: because `%s` call `.String()`

Comment (emilevauge) [inline cmd/traefik/traefik.go]: OK then ;)

**Pull request merged**